Assets/Scripts/SpawnTower.cs

```diff
@@ -21,28 +21,42 @@ private void Start()
 
     private void FixedUpdate()
     {
-        PlaceTower(objToSpawn);
+        PlaceTower();
     }
 
-    public void PlaceTower(GameObject tower)
+    public void PlaceTower()
     {
         _myTime += Time.deltaTime;
         // waits for 1 Second before allowing to spawn another Tower
         SetDisplacement();
-        var neededResources = createDic(tower.GetComponent<NecessaryResources>());
+        var neededResources = createDic(_towerToSpawn.GetComponent<NecessaryResources>());
 
-        if (Input.GetKey(KeyCode.E) && ResourceManager.HasResources(neededResources)
-                                    && SpawnObject.AllowedToSpawn(tower, _objSpawnPlace.position + _displacement)
-                                    && _myTime > _nextPlacement)
+        if (Input.GetKey(KeyCode.E))
         {
-            ResourceManager.UseResource(neededResources);
-            _nextPlacement = _myTime + placementRate;
-            SpawnObject.Spawn(tower, _objSpawnPlace.position + _displacement, _objSpawnPlace.rotation);
-            _nextPlacement -= _myTime;
-            _myTime = 0F;
+            Debug.Log("E");
+            if (canPlaceTower())
+            {
+                Debug.Log("placeTower");
+                ResourceManager.UseResource(neededResources);
+                _nextPlacement = _myTime + placementRate;
+                SpawnObject.Spawn(_towerToSpawn, _objSpawnPlace.position + _displacement, _objSpawnPlace.rotation);
+                _nextPlacement -= _myTime;
+                _myTime = 0F;
+            }
         }
     }
 
+    private bool canPlaceTower()
+    {
+        var neededResources = createDic(_towerToSpawn.GetComponent<NecessaryResources>());
+        return SpawnObject.AllowedToSpawn(_towerToSpawn, _objSpawnPlace.position + _displacement) && ResourceManager.HasResources(neededResources) && _myTime > _nextPlacement;
+    }
+
+    public void setTower(GameObject tower)
+    {
+        _towerToSpawn = tower;
+    }
+
     //TODO solve in NecessaryResources
     private Dictionary<string, int> createDic(NecessaryResources neededResources)
```

Comment:
unrelated to you  - but it seems easier to just refactor ResourceManger 
can you update the TODO?

Reply:
no

Reply:
Mean but fine